v3/library-after-success.sh

```diff
@@ -14,16 +24,31 @@ if [[ "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then
   # Only run releases during a git tag build.
   if [[ -n "$TRAVIS_TAG" ]]; then
 
-    # Allow package.json to specify a custom build script.
-    if npm run | grep -q build:ci; then
-      echo -e "Running custom build step... `npm run build:ci`";
-      npm run build:ci
+    # Install the TravisCI CLI so we can easily make API calls.
+    gem install travis
+
+    # Read the uppercase GitHub username that initiated the build.
+    CREATED_BY=$(travis raw /v3/build/$TRAVIS_BUILD_ID --json --skip-completion-check | jq -r '.created_by.login' | tr 'a-z' 'A-Z' )
+
+    # The spaces here are extremely important as they stop false positives.
+    # For example, without them, "user" would match "username".
+    if [[ " ${SKYUX_TEAM[@]} " =~ " ${CREATED_BY} " ]]; then
+      echo -e "${CREATED_BY} has permission to release."
+
+      # Allow package.json to specify a custom build script.
+      if npm run | grep -q build:ci; then
+        echo -e "Running custom build step... `npm run build:ci`";
+        npm run build:ci
+      else
+        skyux build-public-library
+      fi
+
+      cd dist
+      bash <(curl -s https://blackbaud.github.io/skyux-travis/after-success.sh)
+
     else
-      skyux build-public-library
+      echo -e "${CREATED_BY} lacks permission to release.  Please contact the SKY UX team."
```

Comment:
Since this check is only made on a tag commit, I feel like it should fail the build step, instead of just a warning, so that we aren’t confused (the build is green, but the release didn’t go through). What do you think?

Reply:
Since this is happening in `after-success`, it's too late in the build process to fail it.  This same problem exists today if the `npm publish` step fails.

Reply:
Ah, dang. Okay then!